### PR TITLE
feat: Add fieldChildren method to napi's SgNode and pyo3's PyNode

### PR DESCRIPTION
**Pull request by mohebifar**
`napi`'s SgNode currently only exposes a `field` method that returns the first node matching the given field name. However, there are some cases where a node can have multiple children with the same field. For example, a `jsx_element` in TSX has multiple `attribute` fields.

Screenshot 2024-12-10 at 10 02 13 PM

This new `fieldChildren` returns an array of all the matching nodes.

## Summary by CodeRabbit

- **New Features**
	- Added a new method to retrieve all child nodes in a specified field.
- **Documentation**
	- Updated documentation for existing methods to clarify their functionality.
- **Tests**
	- Introduced a new test case to verify the functionality of the new method.

### CODE REVIEW

**Comment by HerringtonDarkholme** [comment]
Thanks. Can you also port this change to Python API?

**Comment by mohebifar** [comment]
Will do now.

File: crates/pyo3/src/py_node.rs [hunk 1]
```diff
@@ -168,6 +168,17 @@ impl SgNode {
     })
   }
 
+  fn field_children(&self, name: &str) -> Vec<SgNode> {
```
Comment (HerringtonDarkholme) [inline crates/pyo3/src/py_node.rs:171]: Thanks! Another place we need to add is the pyi file. https://github.com/ast-grep/ast-grep/blob/efba7e5f30540f816bfa759ae4192b444d90f8d0/crates/pyo3/ast_grep_py/ast_grep_py.pyi#L56

**Review by HerringtonDarkholme — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**